waku/waku_store/rpc_codec.nim

```diff
@@ -3,257 +3,210 @@ when (NimMajor, NimMinor) < (1, 4):
-  pb.write3(4, index.pubsubTopic)
-  pb.finish3()
-
-  pb
-
-proc decode*(T: type PagingIndexRPC, buffer: seq[byte]): ProtobufResult[T] =
-  ## creates and returns an Index object out of buffer
-  var rpc = PagingIndexRPC()
-  let pb = initProtoBuffer(buffer)
-
-  var data: seq[byte]
-  if not ?pb.getField(1, data):
-    return err(ProtobufError.missingRequiredField("digest"))
-  else:
-    var digest = MessageDigest()
-    for count, b in data:
-      digest.data[count] = b
-
-    rpc.digest = digest
-
-  var receiverTime: zint64
-  if not ?pb.getField(2, receiverTime):
-    return err(ProtobufError.missingRequiredField("receiver_time"))
-  else:
-    rpc.receiverTime = int64(receiverTime)
-
-  var senderTime: zint64
-  if not ?pb.getField(3, senderTime):
-    return err(ProtobufError.missingRequiredField("sender_time"))
-  else:
-    rpc.senderTime = int64(senderTime)
-
-  var pubsubTopic: string
-  if not ?pb.getField(4, pubsubTopic):
-    return err(ProtobufError.missingRequiredField("pubsub_topic"))
-  else:
-    rpc.pubsubTopic = pubsubTopic
-
-  ok(rpc)
-
-proc encode*(rpc: PagingInfoRPC): ProtoBuffer =
-  ## Encodes a PagingInfo object into a ProtoBuffer
-  ## returns the resultant ProtoBuffer
-  var pb = initProtoBuffer()
-
-  pb.write3(1, rpc.pageSize)
-  pb.write3(2, rpc.cursor.map(encode))
-  pb.write3(
-    3,
-    rpc.direction.map(
-      proc(d: PagingDirection): uint32 =
-        uint32(ord(d))
-    ),
-  )
-  pb.finish3()
-
-  pb
-
-proc decode*(T: type PagingInfoRPC, buffer: seq[byte]): ProtobufResult[T] =
-  ## creates and returns a PagingInfo object out of buffer
-  var rpc = PagingInfoRPC()
-  let pb = initProtoBuffer(buffer)
-
-  var pageSize: uint64
-  if not ?pb.getField(1, pageSize):
-    rpc.pageSize = none(uint64)
-  else:
-    rpc.pageSize = some(pageSize)
-
-  var cursorBuffer: seq[byte]
-  if not ?pb.getField(2, cursorBuffer):
-    rpc.cursor = none(PagingIndexRPC)
-  else:
-    let cursor = ?PagingIndexRPC.decode(cursorBuffer)
-    rpc.cursor = some(cursor)
-
-  var direction: uint32
-  if not ?pb.getField(3, direction):
-    rpc.direction = none(PagingDirection)
-  else:
-    rpc.direction = some(PagingDirection(direction))
-
-  ok(rpc)
-
-## Wire protocol
-
-proc encode*(rpc: HistoryContentFilterRPC): ProtoBuffer =
-  var pb = initProtoBuffer()
-
-  pb.write3(1, rpc.contentTopic)
-  pb.finish3()
-
-  pb
+  pb.write3(1, req.requestId)
+  pb.write3(2, req.includeData)
 
-proc decode*(T: type HistoryContentFilterRPC, buffer: seq[byte]): ProtobufResult[T] =
-  let pb = initProtoBuffer(buffer)
-
-  var contentTopic: ContentTopic
-  if not ?pb.getField(1, contentTopic):
-    return err(ProtobufError.missingRequiredField("content_topic"))
-  ok(HistoryContentFilterRPC(contentTopic: contentTopic))
+  pb.write3(3, req.pubsubTopic)
 
-proc encode*(rpc: HistoryQueryRPC): ProtoBuffer =
-  var pb = initProtoBuffer()
-  pb.write3(2, rpc.pubsubTopic)
+  for contentTopic in req.contentTopics:
+    pb.write3(4, contentTopic)
```

Comment:
Truly not sure how to keep track of this in future use of proto3 (other than to create our own library wrapper), but for proto3 we have to use `writePacked` for encoding repeated fields and `getPackedRepeatedField` for decoding. See https://github.com/waku-org/nwaku/pull/2509

Reply:
Ah yes. I remember vaguely something about compat. being broken otherwise.

Reply:
`[packed = true] can only be specified for repeated primitive fields.`

This is the error that @richard-ramos found while trying to pack repeated strings.